egret/model_library/transmission/branch.py

```diff
@@ -785,11 +792,68 @@ def declare_ineq_p_branch_thermal_lbub(model, index_set,
             if p_thermal_limits[branch_name] is None:
                 continue
 
-            m.ineq_pf_branch_thermal_lb[branch_name] = \
-                -p_thermal_limits[branch_name] <= m.pf[branch_name]
-
-            m.ineq_pf_branch_thermal_ub[branch_name] = \
-                m.pf[branch_name] <= p_thermal_limits[branch_name]
+            if neg_slacks and branch_name in m.pf_slack_neg:
+                pf_bn = m.pf[branch_name]
+                if hasattr(pf_bn, 'expr') and isinstance(pf_bn.expr, LinearExpression):
+                    ## create a copy
+                    old_expr = pf_bn.expr
+                    expr = LinearExpression(constant=old_expr.constant,
+                                            linear_vars = old_expr.linear_vars[:] + [m.pf_slack_neg[branch_name]],
+                                            linear_coefs = old_expr.linear_coefs[:] + [1],
+                                            )
+
+                else:
+                    expr = m.pf[branch_name] + m.pf_slack_neg[branch_name]
+                m.ineq_pf_branch_thermal_lb[branch_name] = \
+                    (-p_thermal_limits[branch_name], expr, None)
+            else:
+                m.ineq_pf_branch_thermal_lb[branch_name] = \
+                    (-p_thermal_limits[branch_name], m.pf[branch_name], None)
+
+            if pos_slacks and branch_name in m.pf_slack_pos:
+                pf_bn = m.pf[branch_name]
+                if hasattr(pf_bn, 'expr') and isinstance(pf_bn.expr, LinearExpression):
+                    ## create a copy
+                    old_expr = pf_bn.expr
+                    expr = LinearExpression(constant=old_expr.constant,
+                                            linear_vars = old_expr.linear_vars[:] + [m.pf_slack_pos[branch_name]],
+                                            linear_coefs = old_expr.linear_coefs[:] + [-1],
+                                            )
+                else:
+                    expr = m.pf[branch_name] - m.pf_slack_pos[branch_name]
+                m.ineq_pf_branch_thermal_lb[branch_name] = \
+                    (None, expr, p_thermal_limits[branch_name])
+            else:
+                m.ineq_pf_branch_thermal_ub[branch_name] = \
+                    (None, m.pf[branch_name], p_thermal_limits[branch_name])
+
+def generate_thermal_bounds(pf, llimit, ulimit, neg_slack=None, pos_slack=None):
+    if hasattr(pf, 'expr') and isinstance(pf.expr, LinearExpression):
```

Comment:
Can you add a doc string to this function. The purpose of it is not clear.

Reply:
Done!